[ArsTechnicaBridge] Fix the bridge after redesign

SqrtMinusOne opened the pull request.

SqrtMinusOne committed:
[ArsTechnicaBridge] Fix the bridge after redesign

SqrtMinusOne committed:
[ArsTechnicaBridge] Fix lint

SqrtMinusOne commented:
https://termpad.com is down, it seems.

dvikan commented:
careful changing `uid` because feed readers might produce duplicates of same item

SqrtMinusOne committed:
[ArsTechnicaBridge] Fix uid

SqrtMinusOne committed:
[ArsTechnicaBridge] UID has to be string

SqrtMinusOne commented:
Fixed that. The URL looked like this in the previous version: https://arstechnica.com/?p=2046636

Now the URLs are full, but the post ID can still be found in the page metadata.

SqrtMinusOne commented:
Just 4 hours after me! https://github.com/RSS-Bridge/rss-bridge/pull/4283

@Phantop I suggest we decide which PR to merge. I think I've handled some cases your version doesn't, namely:
- added the subheader and the entry image to content;
- fixed these new reactive galleries; now images are properly wrapped in `<figure>`, `<figcaption>` and nothing else;
- fixed YouTube iframes;
- and the above-mentioned uids.

But feel free to correct.

Phantop commented:
Ah! Yours seems much more in-depth. Not at my  computer to check, but as long as the full content is grabbed without unnecessary elements I don't mind at all! I just sorta did the bare minimum in my PR to get the bridge going again.

Phantop commented:
Had the chance to take a more proper look—closed my PR, as this is definitely the more substantial option and I think should be merged over my work. I do wonder, why is the lightbox fix necessary? Do some readers render that content oddly? Also, can't believe I missed the post ID in the parsely JSON... I did check but managed not to notice.

One small comment: consider removing elements matching the selector `.toc-container`, as certain very long articles (e.g. the macOS Sequoia review) include those but they don't work without the website's Javascript.

SqrtMinusOne committed:
[ArsTechnicaBridge] Remove .toc-container

SqrtMinusOne commented:
@Phantop Done, thanks for the suggestion. @dvikan I think the PR can be merged then.

The lightbox fix is necessary because of how Ars implemented reactivity, e.g., see this article: https://arstechnica.com/gaming/2024/10/halls-of-torment-is-diablo-cranked-up-to-50000-kills-hour/?itm_source=parsely-api. The same text is repeated multiple times:

<details>
  <summary>HTML</summary>

(I've commented something out)

```html
<div class="ars-lightbox align-fullwidth my-5">
  <div class="flex flex-col flex-nowrap gap-5 py-5 md:flex-row">
    <div class="" style="flex-basis: calc(49.152542372881% - 10px);">
      <div class="ars-lightbox-item relative block h-full w-full overflow-hidden rounded-sm">
        <!-- Image 1 -->
        <div class="pswp-caption-content" id="caption-2054558">
          <!-- Caption for image 1 -->
        </div>
      </div>
      <div class="md:hidden">
        <!-- Caption for image 1 -->
      </div>
    </div>
    <div class="flex-1">
      <div class="ars-lightbox-item relative block h-full w-full overflow-hidden rounded-sm">
        <!-- Image 2 -->
        <div class="pswp-caption-content" id="caption-2054552">
          <!-- Caption for image 2 -->
        </div>
      </div>
      <div class="md:hidden">
        <!-- Caption for image 2 -->
      </div>
    </div>
  </div>
  <div class="hidden md:block">
    <div class="ars-gallery-caption">
      <div class="ars-gallery-caption-content left">
        <!-- Caption for image 1 -->
      </div>
    </div>
    <div class="ars-gallery-caption">
      <div class="ars-gallery-caption-content right">
        <!-- Caption for image 2 -->
      </div>
    </div>
  </div>
</div>

```
</details>


The CSS is configured to show different divs on different screen sizes. Without it, all three captions are visible at the same time. Also, when CSS isn't an option, images look better with `figure` and `figcaption`. That at least gives everything proper padding.

So I've figured it would be more straightforward to rewrite the whole thing than to patch the existing one.
